src/test/java/org/jabref/gui/edit/CopyMoreActionTest.java

```diff
@@ -0,0 +1,168 @@
+                .withField(StandardField.TITLE, title)
+                .withField(StandardField.YEAR, "2020")
+                .withField(StandardField.DOI, "10.1145/3377811.3380330")
+                .withField(StandardField.SUBTITLE, "cognitive biases and software development")
+                .withCitationKey("abc");
+        titles.add(title);
+        keys.add("abc");
+    }
+
+    @Test
+    public void testExecuteOnFail() {
+        when(stateManager.getActiveDatabase()).thenReturn(Optional.empty());
+        when(stateManager.getSelectedEntries()).thenReturn(FXCollections.emptyObservableList());
+        copyMoreAction = new CopyMoreAction(StandardActions.COPY_TITLE, dialogService, stateManager, clipBoardManager, preferencesService);
+        copyMoreAction.execute();
+
+        verify(clipBoardManager, times(0)).setContent(any(String.class));
+        verify(dialogService, times(0)).notify(any(String.class));
+    }
+
+    @Test
+    public void testExecuteCopyTitleWithNoTitle() {
+        BibEntry entryWithNoTitle = (BibEntry) entry.clone();
+        entryWithNoTitle.clearField(StandardField.TITLE);
+        ObservableList<BibEntry> entriesWithNoTitles = FXCollections.observableArrayList(entryWithNoTitle);
+        BibDatabaseContext databaseContext = new BibDatabaseContext(new BibDatabase(entriesWithNoTitles));
+
+        when(stateManager.getActiveDatabase()).thenReturn(Optional.ofNullable(databaseContext));
+        when(stateManager.getSelectedEntries()).thenReturn(entriesWithNoTitles);
+        copyMoreAction = new CopyMoreAction(StandardActions.COPY_TITLE, dialogService, stateManager, clipBoardManager, preferencesService);
+        copyMoreAction.execute();
+
+        verify(clipBoardManager, times(0)).setContent(any(String.class));
+        verify(dialogService, times(1)).notify(Localization.lang("None of the selected entries have titles."));
+    }
+
+    @Test
+    public void testExecuteCopyTitleOnPartialSuccess() {
+        BibEntry entryWithNoTitle = (BibEntry) entry.clone();
+        entryWithNoTitle.clearField(StandardField.TITLE);
+        ObservableList<BibEntry> mixedEntries = FXCollections.observableArrayList(entryWithNoTitle, entry);
+        BibDatabaseContext databaseContext = new BibDatabaseContext(new BibDatabase(mixedEntries));
+
+        when(stateManager.getActiveDatabase()).thenReturn(Optional.ofNullable(databaseContext));
+        when(stateManager.getSelectedEntries()).thenReturn(mixedEntries);
+        copyMoreAction = new CopyMoreAction(StandardActions.COPY_TITLE, dialogService, stateManager, clipBoardManager, preferencesService);
+        copyMoreAction.execute();
+
+        String copiedTitles = String.join("\n", titles);
+        verify(clipBoardManager, times(1)).setContent(copiedTitles);
+        verify(dialogService, times(1)).notify(Localization.lang("Warning: %0 out of %1 entries have undefined title.",
```

Comment:
Yes, any()

Reply:
Yes, they can by just any() but because each test is testing for a specific condition, for example when no entries have titles, or when some entries have titles. In different cases, the diaglogService notitfy shows different corresponding messages. I thought it would be better to specify the messages that we are verifying, otherwise the verify dialogService kind of loses its purpose if we us any() as the argument?

Reply:
I think it's okay to test the correct invocation. I think @calixtus  and I thought verify only check with times the number of invovations.